docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+  :ref:`certificate authority <glossCA>` and preconfigures clients and servers
+  to trust this CA.
+
+* **Generated self-signed certificates** — A certificate generated just for
+  Trino that is not automatically trusted by any client. Before using, make sure
+  you understand the :ref:`limitations of self-signed certificates
+  <self_signed_limits>`.
+
+The most convenient option is to purchase a PEM certificate from a Certificate
+Authority that is already listed in the Java truststore. This case requires no
+extra configuration for clients to *just work* when connecting to your server.
+
+Keys, certificates, and containers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Trino recognizes public and private :ref:`keys <glossKey>` as specified in the
+PKCS #1 standard, and recognizes X.509 :ref:`certificates <glossCert>`. Trino
+supports keys and certificates referenced directly as PEM-encoded files without
+a container, which is the preferred method and easiest to work with.
+
+Trino also supports keys and certificates imported into legacy
+:ref:`JKS <glossJKS>` keystore containers, as well as those imported into the
+binary :ref:`PKCS #12 <glossPKCS12>` archive file format.
+
+Make sure you obtain a certificate that is validated by a recognized
+:ref:`certificate authority <glossCA>`.
+
+Inspect received certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Be sure to inspect and validate your received key and certificate files to make
+sure they reference the correct information to access your Trino server. Much
+unnecessary debugging time is saved by taking the time to validate your
+certificates before proceeding to configure the server.
+
+Inspect PEM-encoded files as described in :doc:`Inpsect PEM files
+</security/inspect-pem>`.
+
+Inspect JKS keystores as described in :doc:`Inspect JKS files
+</security/inspect-jks>`.
+
+Invalid certificates
+^^^^^^^^^^^^^^^^^^^^^
+
+If your certificate does not pass validation, or does not show the expected
+information on inspection, contact the group or vendor who provided it for a
+replacement.
+
+.. _cert-placement:
+
+Place the certificate file
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are no location requirements for a certificate file as long as:
+
+* The location is visible from the server's configuration file location with a
+  relative path reference from the server's root directory, or with an absolute
+  path reference.
+* The location is secure from copying or tampering by malicious actors.
+
+You can place your file in the Trino server's ``etc`` directory, which allows
+you to use a relative path reference in configuration files. However, this
+location can require you to keep track of the certificate file, and move it to
+a new ``etc`` directory when you upgrade your Trino version.
+
+.. _configure-https:
+
+Configure the coordinator
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+On the coordinator, add the following lines to the :ref:`config properties file
+<config_properties>` to enable HTTPS support for the server.
```

Comment:
Maybe add a sentence saying, for HTTPS on workers see the internal communications documentation.

Reply:
OK